payments-core/src/main/java/com/stripe/android/CreateIntentCallback.kt

```diff
@@ -5,12 +5,29 @@ import androidx.annotation.RestrictTo
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 sealed interface AbsCreateIntentCallback
 
+/**
+ * Callback to be used when you use `PaymentSheet` and intend to create the [PaymentIntent] or
+ * [SetupIntent] on your server.
+ */
 @ExperimentalPaymentSheetDecouplingApi
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 fun interface CreateIntentCallback : AbsCreateIntentCallback {
 
+    /**
+     * Called when the customer confirms the payment or setup.
+     *
+     * Your implementation should create a [PaymentIntent] or [SetupIntent] on your server and
+     * return its client secret or an error if one occurred.
+     *
+     * @param paymentMethodId The ID of the [PaymentMethod] representing the customer's payment
+     * details
+     */
     suspend fun onCreateIntent(paymentMethodId: String): Result
 
+    /**
+     * Represents the result of a [CreateIntentCallback] or
+     * [CreateIntentCallbackForServerSideConfirmation].
```

Comment:
Should this result type be outside this type, since it's used in both?

Reply:
That sounds reasonable to me, especially when we add further callback types (which it seems like we will be doing). It could become `CreateIntentResult`?

Reply:
Works for me. Should we organize this stuff into a package? (I'm not sure it's the right thing to do, just pondering).